cfgrammar/src/lib/header.rs

```diff
@@ -345,6 +347,132 @@ impl<'input> GrmtoolsSectionParser<'input> {
     }
 }
 
+#[derive(Debug, Eq, PartialEq)]
+pub struct Header {
+    contents: MarkMap<String, (Location, Value)>,
+}
+
+impl Header {
+    pub fn new() -> Self {
+        Header {
+            contents: MarkMap::new(),
+        }
+    }
+
+    pub fn merge_from(
```

Comment:
We'll need to document these functions since they're part of the public interface.

Reply:
I partially did this in [c9756c8](https://github.com/softdevteam/grmtools/pull/549/commits/c9756c8fcadd82f6cb6784c3e83218c02368bf3f)

Then a did a pass over [f671bd4](https://github.com/softdevteam/grmtools/pull/549/commits/f671bd45aa03462cee260b05591bc26f9921624d) since the first commit references `MarkMap`, but `merge_from` was already documented there.